test/array_family/test_reflection_table.py

```diff
@@ -1450,3 +1450,85 @@ def test_calculate_entering_flags(dials_regression):
     flags = refl["entering"]
     assert flags.count(True) == 58283
     assert flags.count(False) == 57799
+
+
+def test_match_1():
+    """Test basic reflection matching"""
+
+    n = 100
+    s = 10
+
+    def r():
+        return random.random()
+
+    xyz = flex.vec3_double()
+
+    for j in range(n):
+        xyz.append((r() * s, r() * s, r() * s * 20))
+
+    order = list(range(n))
+
+    random.shuffle(order)
+
+    xyz2 = xyz.select(flex.size_t(order))
+
+    a = flex.reflection_table()
+    b = flex.reflection_table()
+
+    a["xyz"] = xyz
+    b["xyz"] = xyz2
+
+    mm, nn, distance = a.match(b, key="xyz", scale=(1.0, 1.0, 0.05))
+
+    # FIXME clean this once the change set from cctbx/cctbx_project#581 is in
+    # the stable branch
+    if hasattr(mm, "as_size_t"):
+        a_ = a.select(mm.as_size_t())
+        b_ = b.select(nn.as_size_t())
+    else:
+        a_ = a.select(flex.size_t(list(mm)))
+        b_ = b.select(flex.size_t(list(nn)))
+
+    for _a, _b in zip(a_, b_):
+        assert pytest.approx(_a, _b)
+
+
+def test_match_2():
+    """Test non-1:1 reflection matching"""
+
```

Comment:
This is kind of a personal preference but I think descriptive test names are better than numeric suffixes.

```suggestion
def test_match_mismatched_sizes():
```